app/javascript/src/ChangePassword/components/ChangePassword.jsx

```diff
@@ -18,25 +18,109 @@ type Props = {
   errors: (?FlashMessage)[]
 }
 
+const PASSWORD = 'user[password]'
+const PASSWORD_CONFIRMATION = 'user[password_confirmation]'
+
 const validationConstraints = {
-  'user[password]': {
+  [PASSWORD]: {
     presence: true,
     length: { minimum: 1 }
   },
-  'user[password_confirmation]': {
+  [PASSWORD_CONFIRMATION]: {
     presence: true,
-    length: { minimum: 1 }
+    length: { minimum: 1 },
+    equality: PASSWORD
   }
 }
 
-const ChangePassword = ({ lostPasswordToken, url, errors }: Props) => {
-  const [isFormDisabled, setIsFormDisabled] = useState(true)
+const isValidForm = (form) => validateForm(form, validationConstraints)
+
+const useFormState = () => {
+  const [isDisabled, setIsDisabled] = useState(true)
 
   const onChange = (event) => {
-    const errors = validateForm(event.currentTarget, validationConstraints)
-    setIsFormDisabled(!!errors)
+    const errors = isValidForm(event.currentTarget)
+    setIsDisabled(!!errors)
+  }
+
+  return {
+    isDisabled,
+    onChange
+  }
+}
+
+const usePasswordState = () => {
+  const [value, setValue] = useState('')
+  const [isValid, setIsValid] = useState(undefined)
+
+  const onChange = (value, event) => {
+    setValue(value)
+    setIsValid(!!validateSingleField(event))
+  }
+
+  const onBlur = (event) => {
+    const errors = validateSingleField(event)
+    setIsValid(!!errors)
+  }
+
+  return {
+    value,
+    isValid,
+    onChange,
+    onBlur
+  }
+}
+
+const getPasswordConfirmationErrors = (event) => {
+  const errors = isValidForm(event.currentTarget.parentNode.parentNode)
+  const passwordConfirmationError = (errors && errors[PASSWORD_CONFIRMATION]) && errors[PASSWORD_CONFIRMATION][0]
+  const passwordsMatchError = passwordConfirmationError && passwordConfirmationError.includes('is not equal to')
```

Comment:
is it really necessary to discern this error? Can't just be another password confirmation error?

Reply:
But can't be both treated and rendered the same way? What would make the latter so specific?

Reply:
It should be 0/1: either passwords match or they don't. IMO checking for other errors doesn't really make sense.

Reply:
If everybody agree I can leave a single error: `Password and Password confirmation must match`
WDYT @thomasmaas @cntlsn ?

Reply:
need more context

Reply:
The discussion is if password confirmation field should have 2 possible errors: 
1.- When blank: "Password confirmation is mandatory" 
2.- When it doesn't match with password: "Password and Password confirmation must match"
**OR** just one single error: "Password and Password confirmation must match" for both cases

Reply:
@thomasmaas 
![blank](https://user-images.githubusercontent.com/13486237/75671589-49624380-5c7f-11ea-995e-d6d9081bb052.png)
Vs.
![wrong](https://user-images.githubusercontent.com/13486237/75671596-4d8e6100-5c7f-11ea-8ed0-73ca5732bbe8.png)